### PR TITLE
#719: Displaying the credit card number

### PR DESCRIPTION
**Pull request by hani-ibrahim**
fixing #719

### CODE REVIEW

**Comment by hani-ibrahim** [comment]
👍

File: AtlasUI/AtlasUI/CheckoutSummaryDataModel.swift [hunk 1]
```diff
@@ -106,7 +107,8 @@ extension CheckoutSummaryDataModel {
         self.selectedArticle = selectedArticle
         self.shippingAddress = order.shippingAddress
         self.billingAddress = order.billingAddress
-        self.paymentMethod = checkout?.payment.selected?.method?.localizedTitle
+        let metaData = checkout?.payment.selected?.metadata
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryDataModel.swift]: Inconsistent `Data` capitalization. Above and below, too.

4j5abuy4lmagu

File: AtlasUI/AtlasUI/CheckoutSummaryDataModel.swift [hunk 2]
```diff
@@ -139,11 +143,25 @@ extension CheckoutSummaryDataModel {
...
         default: return Localizer.format(string: "paymentMethod.\(rawValue)")
         }
     }
 
+    private func creditCardLocalizedTitle(metadata: [String: Any]?) -> String {
+        guard let suffix = metadata?["pan_suffix"] as? String else {
+            return Localizer.format(string: "paymentMethod.\(rawValue)")
+        }
+
+        let prefix = metadata?["pan_prefix"] as? String ?? ""
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryDataModel.swift]: Why not add `pan_suffix` and `pan_prefix` in the model?

File: AtlasUI/AtlasUI/CheckoutSummaryDataModel.swift [hunk 3]
```diff
@@ -139,11 +143,25 @@ extension CheckoutSummaryDataModel {
...
+        let availableLength = (prefix + suffix).length
+        let secured = "".padding(toLength: 16 - availableLength, withPad: "x", startingAt: 0)
+        let creditCardNumber = prefix + secured + suffix
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryDataModel.swift]: This whole operation should go to model, too.

File: AtlasUI/AtlasUITests/LoggedInSummaryActionHandlerTests.swift [hunk 4]
```diff
@@ -270,7 +270,7 @@ extension LoggedInSummaryActionHandlerTests {
         return CheckoutSummaryDataModel(selectedArticle: selectedArticle,
                                         shippingAddress: checkout?.shippingAddress,
                                         billingAddress: checkout?.billingAddress,
-                                        paymentMethod: checkout?.payment.selected?.method?.localizedTitle,
+                                        paymentMethod: checkout?.payment.selected?.method?.localizedTitle(metadata: checkout?.payment.selected?.metadata),
```
Comment (bonkey) [inline AtlasUI/AtlasUITests/LoggedInSummaryActionHandlerTests.swift]: And it should be inside payment, too as you're reaching too deep into internals.

Just `checkout?.payment.selected?.localized` or so.

**Comment by bonkey** [comment]
👍

**Comment by hani-ibrahim** [comment]
👍

**Comment by hani-ibrahim** [comment]
creating new branch is much easier than fixing conflicts

**Pull request closed without merging**